From d6c0897154c29c999c7051027e07aede5087b3c2 Mon Sep 17 00:00:00 2001
From: Tomas Henzl <thenzl@redhat.com>
Date: Mon, 8 Sep 2014 14:31:10 -0400
Subject: [scsi] hpsa: complain if physical or logical aborts are not supported

Message-id: <1410186734-19986-20-git-send-email-thenzl@redhat.com>
Patchwork-id: 92076
O-Subject: [RHEL6.6 e-stor PATCH 19/83] hpsa: complain if physical or logical aborts are not supported
Bugzilla: 1069185
RH-Acked-by: Maurizio Lombardi <mlombard@redhat.com>
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Ewan Milne <emilne@redhat.com>

Signed-off-by: Scott Teel <scott.teel@hp.com>
Signed-off-by: Stephen M. Cameron <scameron@beardog.cce.hp.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
(cherry picked from commit 0e7a7fcea03bc527310183c11f4b871c3416d488)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/hpsa.c b/drivers/scsi/hpsa.c
index bd16616..e2277c2 100644
--- a/drivers/scsi/hpsa.c
+++ b/drivers/scsi/hpsa.c
@@ -5082,6 +5082,10 @@ static void hpsa_find_board_params(struct ctlr_info *h)
 
  /* Find out what task management functions are supported and cache */
  h->TMFSupportFlags = readl(&(h->cfgtable->TMFSupportFlags));
+ if (!(HPSATMF_PHYS_TASK_ABORT & h->TMFSupportFlags))
+  dev_warn(&h->pdev->dev, "Physical aborts not supported\n");
+ if (!(HPSATMF_LOG_TASK_ABORT & h->TMFSupportFlags))
+  dev_warn(&h->pdev->dev, "Logical aborts not supported\n");
 }
 
 static inline bool hpsa_CISS_signature_present(struct ctlr_info *h)
-- 
1.7.1